services/roomService/src/lib/CoveyTownController.test.ts

```diff
@@ -158,7 +157,33 @@ describe('CoveyTownController', () => {
       expect(proximityMockListeners[0].onMessageReceived).toBeCalledWith(message);
       expect(proximityMockListeners[1].onMessageReceived).not.toBeCalledWith(message);
     });
+    it('should notify the correct listener when a direct message is sent', async () => {
+      const directMockListeners = [mock<CoveyTownListener>(), mock<CoveyTownListener>()];
+      // player sending message
+      const player1 = new Player('test player 1');
+      // player recieiving direct message
+      const player2 = new Player('test player 2');
+      // player not involved in direct message
+      const player3 = new Player('test player 3');
+      directMockListeners[0].getAssociatedPlayer.mockReturnValue(player2);
+      directMockListeners[1].getAssociatedPlayer.mockReturnValue(player3);
+      testingTown.addPlayer(player2);
+      testingTown.addPlayer(player3);
+      directMockListeners.forEach(listener => testingTown.addTownListener(listener));
 
+      // create a direct message between player 1 and player 2
+      const message = TestUtils.createMessageForTesting(
+        MessageType.DirectMessage,
+        player1,
+        player2,
+      );
+      testingTown.receiveMessage(message);
+
+      // check that player 2 recieved the direct message
+      expect(directMockListeners[0].onMessageReceived).toBeCalledWith(message);
```

Comment:
Again, I think you should create a listener for player1 and make sure that they're also receiving their own message